### PR TITLE
added flatten entites migration script

### PR DESCRIPTION
**Pull request by deepakr123**

### CODE REVIEW

File: module/entities/helper.js [hunk 1]
```diff
@@ -53,15 +54,20 @@ module.exports = class EntitiesHelper {
                 let entityData = await database.models.entities.create(
                     entityDocuments
                 );
+                
+                let entityId = [];
```
Comment (aks30) [inline module/entities/helper.js:58]: @deepakr123 - Rename this to entities variable

File: module/entities/helper.js [hunk 2]
```diff
@@ -1187,8 +1201,119 @@ module.exports = class EntitiesHelper {
         }
     }
 
+     /**
+   * Push entities data to elastic search
+   * @method
+   * @name pushEntitiesToElasticSearch
+   * @name entityId - created or modified entity data.
```
Comment (aks30) [inline module/entities/helper.js:1208]: @deepakr123 - Description doesn't match, also change it to entities

File: module/entities/helper.js [hunk 3]
```diff
@@ -1187,8 +1201,119 @@ module.exports = class EntitiesHelper {
...
+   * @returns {Object} 
+   */
+
+  static pushEntitiesToElasticSearch(entityId) {
```
Comment (aks30) [inline module/entities/helper.js:1212]: @deepakr123 - Rename it to entities, default value to be a blank array

File: module/entities/helper.js [hunk 4]
```diff
@@ -1187,8 +1201,119 @@ module.exports = class EntitiesHelper {
...
+    return new Promise(async (resolve, reject) => {
+        try {
+
+            let telemetry = [];
```
Comment (aks30) [inline module/entities/helper.js:1216]: @deepakr123 - Can be renamed to telemetyEntities, here and in migration

File: module/entities/helper.js [hunk 5]
```diff
@@ -1187,8 +1201,119 @@ module.exports = class EntitiesHelper {
...
+                }, [
+                        "_id",
+                        "metaInformation",
+                        "entityType",
+                        "entityTypeId",
+                        "updatedAt",
+                        "createdAt",
+                    ]);
+            
+            await Promise.all(entityDocuments.map(async entityDocument => {
```
Comment (aks30) [inline module/entities/helper.js:1231]: @deepakr123 - Do it one by one and not in parallel

File: module/entities/helper.js [hunk 6]
```diff
@@ -1187,8 +1201,119 @@ module.exports = class EntitiesHelper {
...
+                    })
+          
+                    entityObj["relatedEntities"] = relatedEntities;
+                  }
+
+                  telemetry.push(telemetryObj);
+
+                  entityObj["telemetry_entities"] = telemetry;
+
+            await elasticSearchData.createOrUpdate(
```
Comment (aks30) [inline module/entities/helper.js:1292]: @deepakr123 - Do format the code with indetation

File: module/entities/helper.js [hunk 7]
```diff
@@ -651,6 +661,7 @@ module.exports = class EntitiesHelper {
     static processEntityMappingUploadData(mappingData = []) {
         return new Promise(async (resolve, reject) => {
             try {
+                let childEntityId = [];
```
Comment (aks30) [inline module/entities/helper.js:664]: @deepakr123 - Rename it to entities please

File: module/entities/helper.js [hunk 8]
```diff
@@ -1187,8 +1201,119 @@ module.exports = class EntitiesHelper {
...
+   * @returns {Object} 
+   */
+
+  static pushEntitiesToElasticSearch(entityId) {
+    return new Promise(async (resolve, reject) => {
+        try {
+
+            let telemetry = [];
+            
+            let entityDocuments = await this.entityDocuments({
```
Comment (aks30) [inline module/entities/helper.js:1218]: @deepakr123 - Perform check of no blank array allowed

File: module/entities/helper.js (hunk 1 above)
Comment (deepakr123) [reply module/entities/helper.js:58]: resolved

File: module/entities/helper.js (hunk 2 above)
Comment (deepakr123) [reply module/entities/helper.js:1208]: resolved

File: module/entities/helper.js (hunk 3 above)
Comment (deepakr123) [reply module/entities/helper.js:1212]: resolved

File: module/entities/helper.js (hunk 4 above)
Comment (deepakr123) [reply module/entities/helper.js:1216]: resolved

File: module/entities/helper.js (hunk 7 above)
Comment (deepakr123) [reply module/entities/helper.js:664]: resolved

File: module/entities/helper.js (hunk 6 above)
Comment (deepakr123) [reply module/entities/helper.js:1292]: resolved

File: module/entities/helper.js (hunk 8 above)
Comment (deepakr123) [reply module/entities/helper.js:1218]: resolved

File: module/entities/helper.js (hunk 5 above)
Comment (deepakr123) [reply module/entities/helper.js]: resolved

**Pull request merged**